daemon/api_snaps_test.go

```diff
@@ -478,6 +478,60 @@ func (s *snapsSuite) TestPostSnapsNoWeirdses(c *check.C) {
 	}
 }
 
+func (s *snapsSuite) TestPostSnapsOptionsUnsupportedActionError(c *check.C) {
+	s.daemon(c)
+	const expectedErr = "snapshot-options can only be specified for snapshot action"
+
+	for _, action := range []string{"install", "refresh", "revert", "remove", "hold", "unhold",
+		"enable", "disable", "switch", "xyzzy"} {
+		holdParams := ""
+		if action == "hold" {
+			holdParams = `"time": "forever", "hold-level": "general",`
+		}
+		buf := strings.NewReader(fmt.Sprintf(`{"action": "%s", "snaps":["foo"], %s "snapshot-options": {}}`, action, holdParams))
+		req, err := http.NewRequest("POST", "/v2/snaps", buf)
+		c.Assert(err, check.IsNil)
+		req.Header.Set("Content-Type", "application/json")
+
+		rspe := s.errorReq(c, req, nil)
+		c.Check(rspe.Status, check.Equals, 400, check.Commentf("%q", action))
+		c.Check(rspe.Message, check.Equals, expectedErr, check.Commentf("%q", action))
+	}
+}
+
+func (s *snapsSuite) TestPostSnapsOptionsOtherErrors(c *check.C) {
+	s.daemon(c)
+	const notListedErr = `cannot use snapshot-options for snap "xyzzy" that is not listed in snaps`
+	const invalidOptionsForSnapErr = `invalid snapshot-options for snap "bar":`
+	const notInstalledErr = `snap "foo" is not installed`
+
+	testMap := map[string]struct {
+		post          string
+		expectedError string
+	}{
+		"snap-not-listed-valid-options": {`{"action": "snapshot", "snaps":["foo", "bar"], "snapshot-options": {"xyzzy": {"exclude":[""]}}}`, notListedErr},
+		"snap-not-listed-exclude-empty": {`{"action": "snapshot", "snaps":["foo", "bar"], "snapshot-options": {"xyzzy": {"exclude":[]}}}`, ""},
```

Comment:
I'm not sure I agree that this should not be an error. Happy to be overriden by Samuele or Miguel here though as I think it can easily argued both ways.

Reply:
sorry, I missed this, yes I think this should be an error, and indeed that might involve reordering how the validation/cleanups  are done

Reply:
@mvo5 I understood your comment to mean that you think that `"exclude":[]` should be an invalid input. Is that right? If so, I don't think the last commit is implementing that but maybe I'm missing the point